src/jit/lowerxarch.cpp

```diff
@@ -1174,13 +1174,13 @@ void Lowering::TreeNodeInfoInitShiftRotate(GenTree* tree)
     else
     {
         MakeSrcContained(tree, shiftBy);
-    }
 
-    // Codegen of shift oper sets ZF and SF flags.
-    // Note that Rotate Left/Right instructions don't set ZF and SF flags.
-    if (tree->OperIsShift())
-    {
-        tree->gtFlags |= GTF_ZSF_SET;
+        // Codegen of shift oper sets ZF and SF flags provided shift count is not zero.
+        // Note that Rotate Left/Right instructions don't set ZF and SF flags.
+        if (tree->OperIsShift() && (shiftBy->gtIntConCommon.IconValue() != 0))
```

Comment:
Note that it is in else part of the if condition on line 1167-1168.  If that condition is false would imply that shiftBy is a constant >= 0 and <=255

---
In reply to: [90954305](https://github.com/dotnet/coreclr/pull/8461#discussion_r90954305) [](ancestors = 90954305)